### DISCUSSION TITLE
How to simplemonitor simplemonitor ?

### DISCUSSION THREAD

**Discussion by wsw70** [jamesoff/simplemonitor#1374] [Q&A] [answered]
After a few years of looking around and as part of the Great Revolution In Self-Hosting And Home Automation 2024 I am coming back to `simplemonitor` because it is the best :) Thank you for this great tool @jamesoff 

I will probably add stuff around Home Assistant (for automated monitoring) once i understand (again) how everything works.

My setup will be
- a few `simplemonitor` instances on the servers at home (probably two) - to monitor the servers themselves and some services
- one instance in the cloud to
  - monitor Internet-facing services
  - retrieve alerts and logging from the internal ones

**My question**: is there a sensible way to monitor `simplemonitor` with `simplemonitor`? :) 
I would like to check from within the LAN that the cloud monitor is still live, so that this cloud monitor can efficiently inform me when the fiber at home goes down. I fear the situation where everything works, the cloud monitor goes down (and I do not know it), then the fiber goes down and I am unaware of that as well.

I have a solid backup monitoring system (my family) but, well, I would prefer to know in advance :)

**Comment by jamesoff** [comment, marked as answer]
I don't actually do this myself, but it's an interesting question. I do have thoughts around a monitor type which knows which remote hosts should be reporting in, and fails if they're not present/fresh (i.e. a remote instance has failed or is otherwise offline).

You could potentially monitor it by having one of the other instances checking if the network port for simplemonitor is open (the tcp monitor), and you can create a configuration which just alerts directly for that monitor but all others are forwarded to the "main" instance instead - I use similar for monitoring my home broadband from a cloud instance. One of my home instances handles alerting for all the others, so that one monitor in the cloud needs to directly alert me if home is offline. I can dig out the config for that if that's of interest.

Hope that helps?

(And thanks for the kind words <3 )

**Reply by wsw70** [reply]
Thank you - this is more or less how I envisioned to do the cross-monitoring.
